### PR TITLE
Test system for running unit tests for Salus.

### PR DESCRIPTION
**Pull request by stillson**
Adds `salus_test` and `run_salus_test` targets. These generate and run a binary under qemu which execute any function label with `#[test_case]` for unit testing. This allows unit testing under riscv emulation.

### CODE REVIEW

File: src/asm.rs [hunk 1]
```diff
@@ -4,4 +4,8 @@
 
 use core::arch::global_asm;
 
+#[cfg(not(test))]
 global_asm!(include_str!("start.S"));
+
+#[cfg(test)]
+global_asm!(include_str!("test_start.S"));
```
Comment (abrestic-rivos) [inline src/asm.rs:11]: do we really need a separate `start.S`?

File: src/main.rs [hunk 2]
```diff
@@ -299,6 +310,28 @@ fn check_vector_width() {
     CSR.sstatus.read_and_clear_bits(sstatus::vs::Dirty.value);
 }
 
+/// The entry point for the test runner
+#[cfg(test)]
+#[no_mangle]
+extern "C" fn test_runner_kernel(_hart_id: u64, _fdt_addr: u64) {
```
Comment (abrestic-rivos) [inline src/main.rs:316]: just call this `kernel_init` and put a `#[cfg(not(test))]` over the other `kernel_init`.

**Review by abrestic-rivos — COMMENTED** [review]
cool!

File: src/asm.rs (hunk 1 above)
Comment (stillson) [reply src/asm.rs:11]: I want to make sure that there is no way for testing code to end up in the shipping binary, and there is no way to use cfg in inline assembly.

File: src/main.rs (hunk 2 above)
Comment (stillson) [reply src/main.rs:316]: in that case, maybe we don't need another start.S

Will fix

**Review by abrestic-rivos — APPROVED** [review]

**Pull request merged**